aws/resource_aws_iot_role_alias.go

```diff
@@ -0,0 +1,144 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iot"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsIotRoleAlias() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIotRoleAliasCreate,
+		Read:   resourceAwsIotRoleAliasRead,
+		Delete: resourceAwsIotRoleAliasDelete,
+		Update: resourceAwsIotRoleAliasUpdate,
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"credential_duration": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      3600,
+				ValidateFunc: validation.IntBetween(900, 3600),
+			},
+		},
+	}
+}
+
+func resourceAwsIotRoleAliasCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+
+	roleAlias := d.Get("alias").(string)
+	roleArn := d.Get("role_arn").(string)
+	credentialDuration := d.Get("credential_duration").(int)
+
+	_, err := conn.CreateRoleAlias(&iot.CreateRoleAliasInput{
+		RoleAlias:                 aws.String(roleAlias),
+		RoleArn:                   aws.String(roleArn),
+		CredentialDurationSeconds: aws.Int64(int64(credentialDuration)),
+	})
+
+	if err != nil {
+		return fmt.Errorf("error creating role alias %s for role %s: %s", roleAlias, roleArn, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s|%s", roleAlias, roleArn))
+	return resourceAwsIotRoleAliasRead(d, meta)
+}
+
+func getIotRoleAliasDescription(conn *iot.IoT, alias string) (*iot.RoleAliasDescription, error) {
+
+	roleAliasDescriptionOutput, err := conn.DescribeRoleAlias(&iot.DescribeRoleAliasInput{
+		RoleAlias: aws.String(alias),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if roleAliasDescriptionOutput == nil {
+		return nil, nil
+	}
+
+	return roleAliasDescriptionOutput.RoleAliasDescription, nil
+}
+
+func resourceAwsIotRoleAliasRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+
+	alias := d.Get("alias").(string)
+
+	var roleAliasDescription *iot.RoleAliasDescription
+
+	roleAliasDescription, err := getIotRoleAliasDescription(conn, alias)
```

Comment:
If this is updated to use `d.Id()` instead of `alias` after updating the resource ID above, we can easily enable importing. 👍 

	roleAliasDescription, err := getIotRoleAliasDescription(conn, d.Id())